src/awkward/_v2/contents/content.py

```diff
@@ -504,6 +504,68 @@ def _localindex_axis0(self):
     def localindex(self, axis):
         return self._localindex(axis, 0)
 
+    def _combinations_axis0(self, n, replacement, recordlookup, parameters):
+        size = len(self)
+        if replacement:
+            size = size + (n - 1)
+        thisn = n
+        combinationslen = 0
+        if thisn > size:
+            combinationslen = 0
+        elif thisn == size:
+            combinationslen = 1
+        else:
+            if thisn * 2 > size:
+                thisn = size - thisn
+            combinationslen = size
+            for j in range(2, thisn + 1):
+                combinationslen = combinationslen * (size - j + 1)
+                combinationslen = combinationslen // j
+
+        nplike_tocarryraw = self.nplike.empty(n, dtype=np.intp)
+        tocarry = []
+        for i in range(n):
+            ptr = ak._v2.index.Index64.empty(
+                combinationslen, self.nplike, dtype=np.int64
+            )
+            tocarry.append(ptr)
+            nplike_tocarryraw[i] = ptr.ptr
+
+        tocarryraw = ak._v2.contents.numpyarray.NumpyArray(nplike_tocarryraw)
+        toindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+        fromindex = ak._v2.index.Index64.empty(n, self.nplike, dtype=np.int64)
+
+        self._handle_error(
+            self.nplike[
+                "awkward_RegularArray_combinations_64",
+                np.int64,
+                toindex.to(self.nplike).dtype.type,
+                fromindex.to(self.nplike).dtype.type,
+            ](
+                tocarryraw.data,
```

Comment:
You don't need to put `nplike_tocarryraw` into `tocarryraw` just to take it out again as `tocarryraw.data`.